packages/react-next/src/components/OverflowSet/OverflowSet.base.tsx

```diff
@@ -1,257 +1,225 @@
           }
-        });
-    } else if (items) {
-      newOverflowItems = items;
+          // Add this persisted keytip to our internal list, use a temporary uniqueID (its content)
+          // uniqueID will get updated on register
+          persistedKeytips[persistedKeytip.content] = persistedKeytip;
+          // Add the overflow sequence to this item
+          const newOverflowItem = {
+            ...overflowItem,
+            keytipProps: {
+              ...keytip,
+              overflowSetSequence: overflowKeytipSequences,
+            },
+          };
+          newOverflowItems.push(newOverflowItem);
+        } else {
+          // Nothing to change, add overflowItem to list
+          newOverflowItems.push(overflowItem);
+        }
+      });
+    } else {
+      newOverflowItems = itemsProp;
     }
-    return <div {...wrapperDivProps}>{this.props.onRenderOverflowButton(newOverflowItems)}</div>;
+    // Set up the keytip effect here, after persistedKeytips
+    useKeytipRegistrations(persistedKeytips, keytipManager);
+    return <div {...wrapperDivProps}>{props.onRenderOverflowButton(newOverflowItems)}</div>;
   };
 
-  /**
-   * Gets the subMenu for an overflow item
-   * Checks if itemSubMenuProvider has been defined, if not defaults to subMenuProps
-   */
+  // Gets the subMenu for an overflow item
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  private _getSubMenuForItem(item: IOverflowSetItemProps): any[] | undefined {
-    if (this.props.itemSubMenuProvider) {
-      return this.props.itemSubMenuProvider(item);
+  const getSubMenuForItem = (item: any): any[] | undefined => {
+    // Checks if itemSubMenuProvider has been defined, if not defaults to subMenuProps
+    if (props.itemSubMenuProvider) {
+      return props.itemSubMenuProvider(item);
     }
     if (item.subMenuProps) {
       return item.subMenuProps.items;
     }
     return undefined;
+  };
+
+  useConst(() => {
+    warnMutuallyExclusive(COMPONENT_NAME, props, {
+      doNotContainWithinFocusZone: 'focusZoneProps',
+    });
+  });
+
+  if (doNotContainWithinFocusZone) {
+    Tag = 'div';
+    uniqueComponentProps = {
+      ...getNativeProps<React.HTMLAttributes<HTMLDivElement>>(props, divProperties),
+      ref: divContainer,
+    };
+  } else {
+    Tag = FocusZone;
+    uniqueComponentProps = {
+      ...getNativeProps<React.HTMLAttributes<HTMLDivElement>>(props, divProperties),
+      ...focusZoneProps,
+      componentRef: focusZone,
+      direction: vertical ? FocusZoneDirection.vertical : FocusZoneDirection.horizontal,
+    };
   }
-}
+
+  useComponentRef(props, focusZone, divContainer);
+
+  return (
+    <Tag
+      role={role || 'group'}
+      aria-orientation={role === 'menubar' ? (vertical === true ? 'vertical' : 'horizontal') : undefined}
+      {...uniqueComponentProps}
+      className={classNames.root}
+    >
+      {overflowSide === 'start' && showOverflow && onRenderOverflowButtonWrapper(overflowItems!)}
```

Comment:
I believe here is wha's causing the hook error. let's move `onRenderOverflowButtonWrapper` to its own FC (call it `OverflowButton`) outside of `OverflowSetBase`